v2_jobs.go

```diff
@@ -80,12 +83,17 @@ func (s *JobStatus) UnmarshalJSON(b []byte) error {
 }
 
 type Task struct {
-	ID       string `json:"uuid"`
-	ExitCode uint8  `json:"exit_code"`
+	ID          string `json:"uuid"`
+	ExitCode    uint8  `json:"exit_code"`
+	Filename    string `json:"file_name"`
+	CreatedAt   string `json:"time_created"`
```

Comment:
I think the date/time values should be represented as a `time.Time` type. I believe `time.Time` supports automatic Unmarshalling of JSON RFC3339 dates: https://eli.thegreenplace.net/2020/unmarshaling-time-values-from-json/

Reply:
You may need to use TaskDateTime (https://pkg.go.dev/go.artefactual.dev/amclient#TaskDateTime) to decode them properly because Archivematica returns a format that time.Time doesn't understand IIRC. I can't remember the details but I think it's because of the leading Z character missing.

Reply:
Huh, if we already have `amclient.TaskDetailed` can't we just use that in jobs struct?  It seems redundant to have the same struct defined in two places.

Reply:
Honestly didn't know there was taskdetailed

Reply:
> I think the date/time values should be represented as a `time.Time` type. I believe `time.Time` supports automatic Unmarshalling of JSON RFC3339 dates: https://eli.thegreenplace.net/2020/unmarshaling-time-values-from-json/

Thought of this. just thinking about time.Time to sql.Null which is what I have to convert this to in preservation task.

Reply:
In `TaskDateTime` you could copy the implementation of [time.UnmarshalJSON()](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/time/time.go;l=1356) and return a nil time if the JSON value is "null" (or an empty string?):

```
func (t *TaskDateTime) UnmarshalJSON(data []byte) error {
        if string(data) == "null" {
             return nil
        }
	s := string(data)
	...
}
```

Then in the preservation task you could convert the `amclient.TaskDateTime` value to a `sql.NullTime` value.

Reply:
Huh, from https://github.com/golang/go/issues/47353 it looks like the `time.UnmarshalJSON()` implementation actually has some issues with escaped Unicode.  Maybe it would be better to implement `UnmarshalText()` instead?

Reply:
> Huh, if we already have `amclient.TaskDetailed` can't we just use that in jobs struct? It seems redundant to have the same struct defined in two places.

	Duration    uint32       `json:"duration"`
task detailed has a duration as a uint32 which it is not. we can create a custom parse function to fix this or create a custom type for it or something call _archivematica_time_ or something haha.